File: 03-Socket-TCP/esempi/es01-maiusc/TcpClient.java
/**
 * Esempio di un semplice client TCP che si connette a un server,
 * invia un messaggio, e riceve una risposta.
 * 
 * Compilo ed eseguo il client:
 * javac TcpClient.java && java TcpClient
 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.net.Socket;

public class TcpClient {
    public static void main(String[] args) {
        try (Socket socket = new Socket("localhost", 8765)) {
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));

            out.println("Ciao dal client!");
            String response = in.readLine();
            System.out.println("Risposta dal server: " + response);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: 03-Socket-TCP/esempi/es01-maiusc/TcpClient.js
/**
 * Esempio di un semplice client TCP che invia un messaggio al server
 * e riceve la risposta.
 *
 * Esecuzione: node TcpClient.js
 */
const net = require('net');

const PORT = 8765;
const HOST = 'localhost';

// Crea il socket client
const client = new net.Socket();

// Connessione al server
client.connect(PORT, HOST, () => {
    console.log(`Connesso al server ${HOST}:${PORT}`);
    
    // Invia un messaggio al server
    const message = 'Ciao dal client Node.js!';
    console.log('Invio messaggio: ' + message);
    client.write(message + '\n');
});

// Gestione dati ricevuti dal server
client.on('data', (data) => {
    console.log('Risposta ricevuta: ' + data.toString().trim());
    client.destroy(); // Chiude la connessione
});

// Gestione chiusura connessione
client.on('close', () => {
    console.log('Connessione chiusa');
});

// Gestione errori
client.on('error', (err) => {
    console.error('Errore:', err.message);
});
